examples/two_mass_spring_damper/run.py

```diff
@@ -0,0 +1,94 @@
+import matplotlib.pyplot as plt
+from numpy import linspace
+from pydy.system import System
+from sympy import symbols, Matrix
+from sympy.physics.mechanics import dynamicsymbols, eombase
+
+# Define the dynamic symbols
+
+x1, x2, u1, u2 = dynamicsymbols('x1 x2 u1 u2')
+
+# Define the constant symbols
+
+m1, c1, k1 = symbols('m1 c1 k1')
+m2, c2, k2 = symbols('m2 c2 k2')
+
+# Define the mass matrix and forcing vector
+
+mm = Matrix([[m1,  0],
+             [0,  m2]])
+f = Matrix([-(k1+k2)*x1 + k2*x2 - (c1+c2)*u1 + c2*u2,
+            k2*x1 - k2*x2 + c2*u1 - c2*u2])
+
+mm_full = Matrix([[1, 0,  0,  0],
+                  [0, 1,  0,  0],
+                  [0, 0, m1,  0],
+                  [0, 0,  0, m2]])
+f_full = Matrix([u1,
+                 u2,
+                 -(k1+k2)*x1 + k2*x2 - (c1+c2)*u1 + c2*u2,
+                 k2*x1 - k2*x2 + c2*u1 - c2*u2])
+
+# Form the rhs of q' = G(q, u, t, r, p). Kinematic equation
+
+G = Matrix([u1, u2])
+
+# Form the rhs of the dynamics equations
+
+RHS = mm_full.inv()*f_full
+
+# Form the remaining input lists for the equations of motion class
+
+coordinates = (x1, x2)
+speeds = (u1, u2)
+states = (x1, x2, u1, u2)
+
+# Form the kinetic energy to attach to one of the EOM instances as an output
+# equation
+
+KE = 0.5 * (m1*u1**2 + m2*u2**2)
+out_eqns = {"kinetic_energy": KE}
+
+# Initialize the equation of motion class using the three forms accepted by
+# ODEFunctionGenerator
+#    [1] x' = F(x, t, r, p)
+#
+#    [2] M(x, p) x' = F(x, t, r, p)
+#
+#    [3] M(q, p) u' = F(q, u, t, r, p)
+#        q' = G(q, u, t, r, p)
+
+eom1 = eombase.EOM(coordinates=coordinates, speeds=speeds, rhs=RHS,
+                   output_eqns=out_eqns)
+eom2 = eombase.EOM(states=states, mass_matrix_full=mm_full, forcing_full=f_full,
+                   num_coordinates=2)
+eom3 = eombase.EOM(states=states, mass_matrix=mm, forcing=f, kinematics=G)
```

Comment:
`Why is num_coordinates needed? And how do you know what order the coordinates are in wrt to the state vector?` 

Number of coordinates is needed for the `sys.plot_coordinates()` method that was discussed last Friday as being a potentially useful method for system. It would be assumed that state is held in the following format [coordinates, speeds, other (ex. Lagrange coefficients)] in which case knowing the number of coordinates the code can pick that many elements off the beginning of the states iterable and have just the coordinates.

`Also, can this work`

This would not be the ideal method because then EOM would have to account for and prepare for every single method that could be used. This would be a maintenance nightmare and make adding new methods an annoying difficult process which is the opposite of the desired result.

`Or should this be how you do it:`

This would be one method. In this way each `*Method` can organize its results into the format that is expected by EOM and would make adding new methods easier as they would simply form the equations of motion their own way then format them for the `EOM` class.

The other way would be `class *Method(eom_base):`. In this method the equations of motion will still need to be specified to the universal API where the base class would provide a blue print. This would require more faith in the programmer of the new `*Method` class to understand and mimic the desired API where the returned object would automatically have the desired API. 

Ex.
```python
class eom_base(object):
    @property
    def mass_matrix(self):
        return self._m_d
```
would require the designers of new classes to include `self._m_d` in their definitions. The advantages of subclassing is that the different components of `*Methods` remain with the equation of motion data (ex. `auxiliary_eqns` or `solve_multipliers()`). For our purposes though I think a returned `EOM` instance might be better as the user can still use the additional information as the `*Method` instance hasn't been deleted, it just isnt what is passed to system.

`I think we need a way to pass in the bodies and loads too.`

I agree this can be useful information.